### PR TITLE
REFACTOR-#3865: remove troubleshooting section about using HDK with pyarrow.gandiva

### LINKED ISSUE #3865: `PyDBEngine` for OmniSci engine and `pyarrow.gandiva` for pyarrow storage format are incompatible
`PyDBEngine` that is being imported when using Modin with `STORAGE_FORMAT=Omnisci` and `pyarrow.gandiva` - when using modin with `STORAGE_FORMAT=Pyarrow` are incompatible:
If ganvida is imported first everything is alright:
```python
import pyarrow.gandiva
from modin.experimental.core.execution.native.implementations.omnisci_on_native.utils import (
    PyDbEngine,
)
```
But vice-versa it fails with segfault:
```python
from modin.experimental.core.execution.native.implementations.omnisci_on_native.utils import (
    PyDbEngine,
)
import pyarrow.gandiva
# Output:
# : CommandLine Error: Option 'enable-vfe' registered more than once!
# LLVM ERROR: inconsistency in registered CommandLine options
# Aborted (core dumped)
```

Packages versions:
```
omniscidb-common          5.8.0                h1234567_1    conda-forge
omniscidbe                5.8.0            h1234567_1_cpu    conda-forge
pyomniscidbe              5.8.0           py38h1234567_1_cpu    conda-forge

arrow-cpp                 5.0.0           py38h43b8b71_5_cpu    conda-forge
pyarrow                   5.0.0           py38h1bc9799_5_cpu    conda-forge
```

Ideally, there shouldn't be scenarios when both of these modules are imported, however, as the recent #3783 showed, it still could happen sometimes.

The role of this issue is to track progress (if any) of resolving the incompatibility of these packages.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3865 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -35,8 +35,6 @@ def noop_decorator(*args, **kwargs):
     sys.modules["cudf"].DataFrame = type("DataFrame", (object,), {})
 if not hasattr(sys.modules["cupy"], "ndarray"):
     sys.modules["cupy"].ndarray = type("ndarray", (object,), {})
-if not hasattr(sys.modules["pyhdk"], "PyDbEngine"):
-    sys.modules["pyhdk"].PyDbEngine = type("PyDbEngine", (object,), {})
```
Comment (anmyachev) [inline docs/conf.py:39]: This is deprecated and is not used anywhere.

**Comment by YarShev** [comment]
The test below still passes, check it out.

https://github.com/modin-project/modin/blob/5e7ce3cd5d02fe98416d9a469baea81718da5eea/modin/test/storage_formats/hdk/test_internals.py#L112

File: docs/conf.py (hunk 1 above)
Comment (Garra1980) [reply docs/conf.py:39]: My understanding that it was kept for backward compatibility, maybe @AndreyPavlenko knows more

Comment (AndreyPavlenko) [reply docs/conf.py:39]: I think it could be safely removed.

**Comment by anmyachev** [comment]
@YarShev this test works on ubuntu but fails on windows.

**Pull request closed without merging**